crates/noirc_frontend/src/parser/parser.rs

```diff
@@ -1470,27 +1471,74 @@ mod test {
             let message = format!("Failed to parse:\n{}", program);
             let (op_t, errors) = parse_recover(&parser, program);
             for e in errors {
-                if !e.is_warrning() {
-                    panic!("{}", &message);
+                if !e.is_warning() {
+                    panic!("{} with error {}", &message, e);
                 }
             }
             op_t.expect(&message)
         })
     }
 
+    fn parse_all_warnings<P, T>(parser: P, programs: Vec<&str>) -> Vec<CustomDiagnostic>
+    where
+        P: NoirParser<T>,
+        T: std::fmt::Display,
+    {
+        programs
+            .into_iter()
+            .flat_map(|program| match parse_recover(&parser, program) {
+                (None, errors) => errors,
+                (Some(expr), errors) => {
+                    let mut has_errors = false;
+                    for e in &errors {
+                        if e.is_error() {
+                            has_errors = true;
+                        }
+                    }
+                    match has_errors {
+                        true => {
+                            unreachable!(
+                                "Expected this input to pass with warning:\n{}\nYet it successfully failed with error:\n{}",
+                                program, expr
+                            )
+                        }
+                        false => {}
+                    }
+                    errors
+
+                }
+            })
+            .collect()
+    }
+
     fn parse_all_failing<P, T>(parser: P, programs: Vec<&str>) -> Vec<CustomDiagnostic>
     where
         P: NoirParser<T>,
         T: std::fmt::Display,
     {
         programs
             .into_iter()
-            .flat_map(|program| match parse_with(&parser, program) {
-                Ok(expr) => unreachable!(
-                    "Expected this input to fail:\n{}\nYet it successfully parsed as:\n{}",
-                    program, expr
-                ),
-                Err(error) => error,
+            .flat_map(|program| match parse_recover(&parser, program) {
```

Comment:
Also, why are we using `parse_recover` now instead of `parse_with`? I'd expect none of our parser test cases would need recovery since we choose the inputs (sans the tests made to explicitly test parser recovery of course).

Reply:
`parse_with` returns successfully parsed program or errors. In case we want to inspect warnings we can't use `parse_with`